scylla/src/transport/connection.rs

```diff
@@ -98,8 +101,21 @@ impl Connection {
 
         let (error_sender, error_receiver) = tokio::sync::oneshot::channel();
 
-        let (fut, _worker_handle) = Self::router(stream, receiver, error_sender).remote_handle();
-        tokio::task::spawn(fut);
+        let _worker_handle = match config.use_tls {
+            true => {
+                let tls_certificate_path = match config.tls_certificate_path {
+                    Some(ref path) => path,
+                    None => panic!("certificate path not provided!"),
+                };
+                let mut connector = SslConnector::builder(SslMethod::tls())?;
+                connector.set_ca_file(tls_certificate_path)?;
```

Comment:
We're going to need more flexibility here. We shouldn't assume that the user wants to use a CA file, they might want to use https://docs.rs/openssl/0.10.32/openssl/ssl/struct.SslContextBuilder.html#method.set_certificate_file , https://docs.rs/openssl/0.10.32/openssl/ssl/struct.SslContextBuilder.html#method.set_certificate_chain_file or https://docs.rs/openssl/0.10.32/openssl/ssl/struct.SslContextBuilder.html#method.set_certificate instead. We should just expose a way for users to provide custom configuration, e.g. by letting them set a custom TLS connector - and then they would be able to set whatever they need.